R/baggr_compare.R

```diff
@@ -136,49 +258,72 @@ baggr_compare <- function(...,
           # geom_jitter(size = 2) +
-          geom_errorbar(size = 1.2, width = 0, position=position_dodge(width=0.5)) +
-          geom_point(size = 2, stroke = 1.5, fill = "white", position=position_dodge(width=0.5), pch = 21) +
-          coord_flip() +
-          labs(x = "", y = "Treatment effect (95% interval)",
-               title = effect_names[i]) +
+          ggplot2::geom_errorbar(size = 1.2, width = 0,
+                                 position = ggplot2::position_dodge(width=0.5)) +
+          ggplot2::geom_point(size = 2, stroke = 1.5, fill = "white",
+                              position = ggplot2::position_dodge(width=0.5),
+                              pch = 21) +
+          ggplot2::coord_flip() +
+          ggplot2::labs(x = "", y = "Treatment effect (95% interval)",
+                        title = paste0(
+                          "Pooled and group-specific effect ",
+                          "of treatment on ",
+                          effect_names[i],
+                          " outcome."
+                          )
+                        ) +
           baggr_theme_get() +
-          theme(legend.position="top")
-        # plot(comparison_plot)
+          ggplot2::theme(legend.position="top")
         return(comparison_plot)
       })
     } else if(compare == "effects"){
-      plots <- do.call(effect_plot, models)
+      plots <- do.call(effect_plot, models) +
+        labs(fill = NULL)
     } else {
-      stop("Argument compare= must be 'effects' or 'groups'.")
+      stop("Argument compare = must be 'effects' or 'groups'.")
     }
   }
 
-  if(length(plots) == 1)
-    plots <- plots[[1]]
+  if("ggplot" %in% class(plots)){
+    return(plots)
+  } else {
+    structure(plots, class = "plot_list")
+  }
+}
 
-  # Print treatment effects
-  mean_trt_effects <- do.call(rbind, (
-    lapply(models, function(x) {
-      mint(treatment_effect(x)$tau)
-    })))
-  sd_trt_effects <- do.call(rbind, (
-    lapply(models, function(x) {
-      mint(treatment_effect(x)$sigma_tau)
-    })))
+#' Print list of baggr plots
+#' @param x list of plots to print
+#' @param ... ignored for now
+#' @details prints plots in a loop, internal use only
+print.plot_list <- function(x) {
+  if(length(x) == 1) {
+    print(x[[1]])
+  } else {
+    for(i in 1:length(plots)) {
+      print(plots[[i]])
+    }
+  }
+}
 
-  cat("Mean treatment effects:\n")
-  print(mean_trt_effects)
-  cat("\n")
-  cat("SD for treatment effects:\n")
-  print(sd_trt_effects)
+#' Make an expression silent
+#' @param ... expression to pass to make silent
+#' @importFrom testthat capture_output
+#' @details Runs the example while suppressing messages and warnings
+#' while sinking the output to a temp file and then deleting
+#' (this bit is handled by testthat)
+#' @examples
+#' make_silent(tmp <- 1:10)
+make_silent <- function(...) {
+  suppressWarnings({
+    suppressMessages({
+      invisible(testthat::capture_output(...))
+    })
+  })
```

Comment:
I'm always wary of suppressWarnings as it can give you trouble if an operation fails. Wasn't `try` enough? (see comment above)

Reply:
I'm adding in a silence_messages option that handles this more directly. Once I push feel free to reject if you feel it is too heavy-handed. I'll come up with something else.